### PR TITLE
Add transformer method for Implicit Anonymous Function Expression Node.

### LINKED ISSUE #23614: Add transformer method for Implicit Anonymous Function Expression.
**Description:**

- $subject

**Suggested Labels**
- Component/Parser

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$subject

Fixes #23614

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 1]
```diff
@@ -1429,6 +1434,42 @@ public BLangNode transform(SyncSendActionNode syncSendActionNode) {
...
+                System.out.println("Empty parameter list");
+            }
+        } else {
+            BLangUserDefinedType userDefinedType = (BLangUserDefinedType) param.apply(this);
+            BLangSimpleVariable parameter = (BLangSimpleVariable) TreeBuilder.createSimpleVariableNode();
+            parameter.name = userDefinedType.typeName;
+            arrowFunction.params.add(parameter);
+        }
+        arrowFunction.body = new BLangExprFunctionBody();
+        arrowFunction.body.expr =  createExpression(implicitAnonymousFunctionExpressionNode.expression());
```
Comment (KavinduZoysa) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1469]: extra space after = sign.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 2]
```diff
@@ -1429,6 +1434,42 @@ public BLangNode transform(SyncSendActionNode syncSendActionNode) {
...
+                SeparatedNodeList<SimpleNameReferenceNode> paramList = paramsNode.parameters();
+
+                for (SimpleNameReferenceNode child : paramList) {
+                    BLangUserDefinedType userDefinedType = (BLangUserDefinedType) child.apply(this);
+                    BLangSimpleVariable parameter = (BLangSimpleVariable) TreeBuilder.createSimpleVariableNode();
+                    parameter.name = userDefinedType.typeName;
+                    arrowFunction.params.add(parameter);
+                }
+            } catch (Exception e) {
+                System.out.println("Empty parameter list");
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1460]: above will fail the style guide.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java [hunk 3]
```diff
@@ -1429,6 +1434,42 @@ public BLangNode transform(SyncSendActionNode syncSendActionNode) {
...
+                ImplicitAnonymousFunctionParameters paramsNode = (ImplicitAnonymousFunctionParameters) param;
+                SeparatedNodeList<SimpleNameReferenceNode> paramList = paramsNode.parameters();
+
+                for (SimpleNameReferenceNode child : paramList) {
+                    BLangUserDefinedType userDefinedType = (BLangUserDefinedType) child.apply(this);
+                    BLangSimpleVariable parameter = (BLangSimpleVariable) TreeBuilder.createSimpleVariableNode();
+                    parameter.name = userDefinedType.typeName;
+                    arrowFunction.params.add(parameter);
+                }
+            } catch (Exception e) {
```
Comment (manuranga) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java:1459]: What is the `Exception` you are trying to catch? we should fix it properly form the parser side first.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 1 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Removed.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 2 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Removed try catch.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java (hunk 3 above)
Comment (rdulmina) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java]: Removed try catch.

**Pull request merged**